mlflow/utils/search_utils.py

```diff
@@ -1102,7 +1102,10 @@ def _does_model_version_match_clauses(cls, mv, sed):  # pylint: disable=argument
 
         # what comparators do we support here?
         if cls.is_string_attribute(key_type, key, comparator):
-            lhs = getattr(mv, key)
+            if key == "source_path":
+                lhs = getattr(mv, key, getattr(mv, "source"))
+            else:
+                lhs = getattr(mv, key)
```

Comment:
`mv` is either `ModelVersion` or `SqlModelVersion`. `SqlModelVersion` has an `source_path` attribute, but `ModelVersion` doesn't.